packages/openapi3/src/openapi.ts

```diff
@@ -1076,6 +1076,7 @@ function createOAPIEmitter(program: Program, options: ResolvedOpenAPI3EmitterOpt
 
     const requestBody: any = {
       description: body.parameter ? getDoc(program, body.parameter) : undefined,
+      required: body.parameter ? !body.parameter.optional : undefined,
```

Comment:
If there is no body parameter but a body. Wouldn’t then the body be required all the time?

Reply:
I didn't realize that the body parameter wasn't present for implicit bodies. I pushed a fix to make implicit bodies required in OpenAPI 3.